### PR TITLE
Security - Stateless JWT with KeyCloak first draft ready for review #55

### PR DESCRIPTION
**Pull request by irinaponomarevahaulmont**
Please review

### CODE REVIEW

File: content/modules/studio/pages/stateless-jwt-authentication.adoc [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+
+Enter `demoadmin/demoadmin` as your username and password. You should see your application now. Log out of your application and try entering non-existent credentials. Keycloak will display an error message.
+
+image::security-jwt/error-message.png[align=center, width=500]
+
+[[setting-up-limitations-by-roles]]
+=== Setting Up Endpoint Access Limitations by Roles
+
+Note that in the generated WebSecurityConfiguration class
+a `JwtAuthenticationConverter` bean is defined among other things.
```
Comment (alexbudarov) [inline content/modules/studio/pages/stateless-jwt-authentication.adoc:313]: things --> elements

File: content/modules/studio/pages/stateless-jwt-authentication.adoc [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+In order to implement this authorization scheme, we need to go to the GraphQL controller and make use of the functionality built into Studio for assigning roles to endpoints. The `delete()` mutation must be accessible only to the users with the `admin` role.
+
+The original code of the mutation is as follows:
+
+[source, java]
+----
+    @MutationMapping(name = "deleteClient")
+    @Transactional
+    public void delete(@GraphQLId @Argument @NonNull Long id) {
+        Client entity = crudRepository.findById(id)
```
Comment (alexbudarov) [inline content/modules/studio/pages/stateless-jwt-authentication.adoc:343]: comment out method body. it's not relevant to our topic.

// ...

File: content/modules/studio/pages/stateless-jwt-authentication.adoc [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+Click on the inlay hint above the `findAll()` method, type ADMIN, USER into the pop-up window (you can usee the `+` button to enter roles that have already been used) and click *_OK_*.
+
+image::security-jwt/admin-user-roles.png[align=center, width=500]
+
+The following annotation will be added:
+
+[source, java]
+@Secured({"ROLE_ADMIN", "ROLE_USER"})
+
+Repeat the same procedure for the remaining two endpoints: `findById()` and `update()`. You can usee the `+` button to enter roles that have already been used.
```
Comment (alexbudarov) [inline content/modules/studio/pages/stateless-jwt-authentication.adoc:374]: > usee

typo

File: content/modules/studio/pages/stateless-jwt-authentication.adoc [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+
+Keycloak is an open source software product to allow single sign-on. Its important features include:
+
+* Identity and access management
+* Social login
+* Two-factor authentication
+* Support of standard protocols
+* Extension support.
+
+This project is aimed at modern applications and services and is under the stewardship of https://www.redhat.com/[Red Hat].
```
Comment (alexbudarov) [inline content/modules/studio/pages/stateless-jwt-authentication.adoc:18]: I would remove this line, too advertisement-ish.

File: content/modules/studio/pages/security.adoc [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+Amplicode tooling supports several authentication types:
+
+* HTTP session authentication (users stored in memory, JPA, etc.)
+* Stateless authentication via a JWT token
+* Stateful OAuth authentication
+* LDAP authentication
+* etc.
+
+Apart from setting up the authentication system, Amplicode tooling allows us set up the following aspects of the application:
```
Comment (alexbudarov) [inline content/modules/studio/pages/security.adoc:13]: authentication system --> authentication scheme

File: content/modules/studio/pages/security.adoc [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+* Configuring authorizations and authentications for Spring security
```
Comment (alexbudarov) [inline content/modules/studio/pages/security.adoc:15]: authorizations and authentications --> authorization and authentication
Spring security --> Spring Security

File: content/modules/studio/pages/security.adoc [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+* Endpoint protection with the help of Spring security roles
```
Comment (alexbudarov) [inline content/modules/studio/pages/security.adoc:16]: Spring security --> Spring Security

**Comment by irinaponomarevahaulmont** [comment]
Requested changes applied, please review.

**Pull request merged**